boards/main/src/communication.rs

```diff
@@ -0,0 +1,212 @@
+use common_arm::mcan::message::Raw;
+use messages::mavlink;
+pub struct CanDevice0 {
+    pub can: Can<'static, Can0, Dependencies<Can0, Gclk0Id, Pin<PA23, Alternate<I>>, Pin<PA22, Alternate<I>>, CAN0>, Capacities>,
+    line_interrupts: OwnedInterruptSet<Can0, EnabledLine0>,
+}
+
+impl CanDevice0 {
+    // we need to give back gclk0
+    pub fn new<S>(
+        can_rx: Pin<PA23, AlternateI>,
+        can_tx: Pin<PA22, AlternateI>,
+        pclk_can: Pclk<Can0, Gclk0Id>,
+        ahb_clock: AhbClk<Can0>,
+        peripheral: CAN0,
+        gclk0: S,
+        can_memory: &'static mut SharedMemory<Capacities>,
+        loopback: bool,
+    ) -> (Self, S::Inc)
+    where
+        S: Source<Id = Gclk0Id> + Increment,
+    {
+        let (can_dependencies, gclk0) =
+            Dependencies::new(gclk0, pclk_can, ahb_clock, can_rx, can_tx, peripheral);
+
+        let mut can =
+            mcan::bus::CanConfigurable::new(200.kHz(), can_dependencies, can_memory).unwrap();
+        can.config().mode = Mode::Fd {
+            allow_bit_rate_switching: false,
+            data_phase_timing: BitTiming::new(500.kHz()),
+        };
+
+        if loopback {
+            can.config().loopback = true;
+        }
+
+        let interrupts_to_be_enabled = can
+            .interrupts()
+            .split(
+                [
+                    Interrupt::RxFifo0NewMessage,
+                    Interrupt::RxFifo0Full,
+                    Interrupt::RxFifo0MessageLost,
+                    Interrupt::RxFifo1NewMessage,
+                    Interrupt::RxFifo1Full,
+                    Interrupt::RxFifo1MessageLost,
+                ]
+                .into_iter()
+                .collect(),
+            )
+            .unwrap();
+
+        // Line 0 and 1 are connected to the same interrupt line
+        let line_interrupts = can
+            .interrupt_configuration()
+            .enable_line_0(interrupts_to_be_enabled);
+
+        can.filters_standard()
+            .push(Filter::Classic {
+                action: Action::StoreFifo0,
+                filter: ecan::StandardId::new(messages::sender::Sender::RecoveryBoard.into()).unwrap(),
+                mask: ecan::StandardId::MAX,
+            })
+            .unwrap_or_else(|_| panic!("Recovery filter"));
+
+        can.filters_standard()
+            .push(Filter::Classic {
+                action: Action::StoreFifo1,
+                filter: ecan::StandardId::new(messages::sender::Sender::GroundStation.into()).unwrap(),
+                mask: ecan::StandardId::MAX,
+            })
+            .unwrap_or_else(|_| panic!("Ground Station filter"));
+
+        let can = can.finalize().unwrap();
+        (
+            CanDevice0 {
+                can,
+                line_interrupts,
+            },
+            gclk0,
+        )
+    }
+    pub fn send_message(
+        &mut self,
+        m: Message,
+    ) -> Result<(), HydraError> {
+        let payload: Vec<u8, 64> = postcard::to_vec(&m)?;
```

Comment:
How does this handle cases where the serialized message is bigger than 64 bytes? It's probably fine for now, but we might want to consider static checks to ensure we don't accidentally make messages too big.

Reply:
The function returns a serialization error if the message is bigger than 64 bytes. Ideally any message defined has been defined to be under 64 bytes like SBGShort.